trillian/integration/ct_integration.go

```diff
@@ -50,17 +50,45 @@ var Verifier = merkletree.NewMerkleVerifier(func(data []byte) []byte {
 	return hash[:]
 })
 
-// ClientPool holds a collection of CT LogClient instances.
-type ClientPool []*client.LogClient
+// ClientPool describes an entity which produces LogClient instances.
+type ClientPool interface {
+	// Next returns the next LogClient instance to be used.
+	Next() *client.LogClient
+}
+
+// RandomPool holds a collection of CT LogClient instances.
+type RandomPool []*client.LogClient
+
+var _ ClientPool = &RandomPool{}
 
-// Pick a random client from the pool.
-func (p ClientPool) Pick() *client.LogClient {
+// Next a random client from the pool.
+func (p RandomPool) Next() *client.LogClient {
```

Comment:
Could you make this take a pointer receiver to match `NewRandomPool` returning a `*RandomPool` as a `ClientPool`.